tests/components/vesync/test_switch.py

```diff
@@ -49,3 +56,75 @@ async def test_switch_state(
     # Check states
     for entity in entities:
         assert hass.states.get(entity.entity_id) == snapshot(name=entity.entity_id)
+
+
+@pytest.mark.parametrize(
+    ("api_response", "expectation"),
+    [(False, pytest.raises(HomeAssistantError)), (True, NoException)],
+)
+async def test_turn_on_display(
+    hass: HomeAssistant,
+    humidifier_config_entry: MockConfigEntry,
+    api_response: bool,
+    expectation,
+) -> None:
+    """Test turn_on method."""
+
+    # turn_on_display returns False indicating failure in which case switch.turn_on_display
+    # raises HomeAssistantError.
+    with (
+        expectation,
+        patch(
+            "pyvesync.vesyncfan.VeSyncHumid200300S.turn_on_display",
+            return_value=api_response,
+        ) as method_mock,
+    ):
+        with patch(
+            "homeassistant.components.vesync.switch.VeSyncSwitchEntity.schedule_update_ha_state"
+        ) as update_mock:
+            await hass.services.async_call(
+                SWITCH_DOMAIN,
+                SERVICE_TURN_ON,
+                {ATTR_ENTITY_ID: ENTITY_SWITCH_DISPLAY},
+                blocking=True,
+            )
+
+        await hass.async_block_till_done()
+        method_mock.assert_called_once()
+        update_mock.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("api_response", "expectation"),
+    [(False, pytest.raises(HomeAssistantError)), (True, NoException)],
+)
+async def test_turn_off_display(
+    hass: HomeAssistant,
+    humidifier_config_entry: MockConfigEntry,
+    api_response: bool,
+    expectation,
+) -> None:
+    """Test turn_off method."""
+
+    # turn_off_display returns False indicating failure in which case switch.turn_off_display
+    # raises HomeAssistantError.
+    with (
+        expectation,
+        patch(
+            "pyvesync.vesyncfan.VeSyncHumid200300S.turn_off_display",
+            return_value=api_response,
+        ) as method_mock,
+    ):
+        with patch(
+            "homeassistant.components.vesync.switch.VeSyncSwitchEntity.schedule_update_ha_state"
+        ) as update_mock:
+            await hass.services.async_call(
+                SWITCH_DOMAIN,
+                SERVICE_TURN_OFF,
+                {ATTR_ENTITY_ID: ENTITY_SWITCH_DISPLAY},
+                blocking=True,
+            )
+
+        await hass.async_block_till_done()
+        method_mock.assert_called_once()
+        update_mock.assert_called_once()
```

Comment:
I think it's more useful and logical to switch the way we parametrize. Have 1 test for all the failing service calls, and have 1 test for all the working service calls, since the only thing you have to change is the function called and the service name.

Reply:
To confirm are you thinking a consolidated test across all platforms so also parametrize a few other values or just within the platform?